src/Controller/ImportExportController.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+namespace App\Controller;
+
+use App\Entity\Contact;
+use Doctrine\ORM\EntityManagerInterface;
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+use PhpOffice\PhpSpreadsheet\Writer\Xlsx;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+use Symfony\Component\Routing\Annotation\Route;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\Extension\Core\Type\FileType;
+use Symfony\Component\Form\Extension\Core\Type\SubmitType;
+
+class ImportExportController extends AbstractController
+{
+    #[Route('/import', name: 'app_import')]
+    function import(Request $request, EntityManagerInterface $entityManager)
+    {
+        $form = $this->createFormBuilder()
+            ->add('submitFile', FileType::class, [
+                'label' => ' '
+            ])
+            ->add('submit', SubmitType::class, [
+                'label' => 'Zatwierdź'
+            ])
+            ->getForm();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var UploadedFile */
+            $file = $form->get('submitFile')->getData();
+            // Otwarcie pliku
+            if (($handle = fopen($file->getPathname(), "r")) !== false) {
+                // Przetwarzanie danych.
+                while (($data = fgetcsv($handle)) !== false) {
+
+                    $query = "INSERT INTO contact (first_name, last_name, company)
```

Comment:
Zapis danych do bazy najlepiej wykonać w osobnej klasie odpowiedzialnej za operowanie na encjach z tabeli contact.
Chodzi o to, że co do zasady klasy miały jak najbardziej ograniczoną odpowiedzialność. To jest pierwsza literka S z SOLID opisującego dobra praktyki w programowaniu obiektowym https://infotraining.bitbucket.io/cpp-tdd/solid-oop.html
Czyli kontroler powinien tylko sterować aplikacją wykonując odpowiednie metody różnych klas. Ale jakiejś większej logiki nie powinno w nim być.

Reply:
Linie 42 - 52 należałoby zastąpić wywołaniem odpowiedniej metody klasy, która będzie potrafiła dodawać rekordy do tabeli contact